src/autocomplete.ts

```diff
@@ -1,20 +1,31 @@
 import { Component } from "./component";
+import { DropdownOptions } from "./dropdown";
 import { M } from "./global";
-import { Dropdown } from "./dropdown";
 
-let _defaults = {
+export interface AutocompleteOptions {
+  data?: any[];
+  onAutocomplete?(selectedValue: any): void;
```

Comment:
this is called as
`this.options.onAutocomplete.call(this, this.selectedValues);`
so it does not need Autocomplete as parameter. 
But other callback as onSearch are called like
`this.options.onSearch(this.el.value, this, this.options);`
so this needs to have the component as a callback parameter. 

I would propose to unify this, and always having the component as parameter for callback methods..
What do you think?